src/Runner.Worker/JobExtension.cs

```diff
@@ -127,6 +128,17 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
                     context.SetRunnerContext("workspace", Path.Combine(_workDirectory, trackingConfig.PipelineDirectory));
                     context.SetGitHubContext("workspace", Path.Combine(_workDirectory, trackingConfig.WorkspaceDirectory));
 
+                    // Temporary hack for GHES alpha
+                    var configurationStore = HostContext.GetService<IConfigurationStore>();
+                    var runnerSettings = configurationStore.GetSettings();
+                    if (!runnerSettings.IsHostedServer && !string.IsNullOrEmpty(runnerSettings.GitHubUrl))
```

Comment:
I think this needs to be `string.IsNullOrEmpty(runnerSettings.GitHubUrl)`.  This will be non-null if you're talking to hosted GitHub -- see [where this is set](https://github.com/actions/runner/blob/dff1024cd3ab8f5b852750be7e1ddc605c3a45da/src/Runner.Listener/Configuration/ConfigurationManager.cs#L98).

Reply:
Or, is that supposed to be an `||` in the middle?

Reply:
good catch

@TingluoHuang when `!runnerSettings.IsHostedServer` should we fallback to `runnerSettings.ServerUrl` when `GitHubUrl` is not defined?

Reply:
runnerSettings.ServerUrl is the pipelines url, pipelines.actions.githubusercontent.com/xxxxx

When GitHubUrl is not defined, it means we connected to Devfabric.

Reply:
ok i think the change here might be fine then, but additional change to figure out for registration against the appliance

Reply:
discussed offline w/ ting, leaving this way for now. When config against appliance works, will need an additional change to persist githuburl for that scenario.